glotaran/project/project.py

```diff
@@ -0,0 +1,446 @@
+        self._result_registry = ProjectResultRegistry(self.folder)
+
+    @classmethod
+    def create(cls, name: str | None = None, folder: str | Path | None = None) -> Project:
+        """Create a new project.
+
+        Parameters
+        ----------
+        name : str | None
+            The name of the project. If ``None``, the name of the project folder will be used.
+        folder : str | Path | None
+            The folder where the project will be created. If ``None``, the current work
+            directory will be used.
+
+        Returns
+        -------
+        Project
+            The created project.
+        """
+        from glotaran import __version__ as gta_version
+
+        if folder is None:
+            folder = getcwd()
+        project_folder = Path(folder)
+        if not project_folder.exists():
+            project_folder.mkdir()
+        name = name or project_folder.name
+        project_file = project_folder / PROJECT_FILE_NAME
+        with open(project_file, "w") as f:
+            f.write(TEMPLATE.format(gta_version=gta_version, name=name))
+
+        return cls.open(project_file)
+
+    @classmethod
+    def open(cls, project_folder_or_file: str | Path):
+        """Open a new project.
+
+        Parameters
+        ----------
+        project_folder_or_file : str | Path
+            The path to a project folder or file.
+
+        Returns
+        -------
+        Project
+            The created project.
+        """
+        folder = Path(project_folder_or_file)
+        if folder.is_dir():
+            file = folder / PROJECT_FILE_NAME
+        else:
+            folder, file = folder.parent, folder
+
+        project_dict = load_dict(file, True)
+        project_dict["file"] = file
+        project_dict["folder"] = folder
+        return cls(**project_dict)
+
+    @property
+    def has_data(self) -> bool:
+        """Check if the project has datasets.
+
+        Returns
+        -------
+        bool
+            Whether the project has datasets.
+        """
+        return not self._data_registry.empty
+
+    @property
+    def data(self) -> dict[str, str]:
+        """Get all project datasets.
+
+        Returns
+        -------
+        dict[str, str]
+            The models of the datasets.
+        """
+        return self._data_registry.items
+
+    def load_data(self, name: str) -> xr.Dataset | xr.DataArray:
+        """Load a dataset.
+
+        Parameters
+        ----------
+        name : str
+            The name of the dataset.
+
+        Returns
+        -------
+        Result
+            The loaded dataset.
+
+        Raises
+        ------
+        ValueError
+            Raised if the dataset does not exist.
+        """
+        try:
+            return self._data_registry.load_item(name)
+        except ValueError:
+            raise ValueError(f"Dataset '{name}' does not exist.")
+
+    def import_data(self, path: str | Path, name: str | None = None):
```

Comment:
Please allow use of [`DatasetMappable`](https://github.com/glotaran/pyglotaran/blob/a7f67d4ad7ec5b45972c5aca3163e234abdc4db0/glotaran/typing/types.py#L11) and use [`load_datasets`](https://github.com/glotaran/pyglotaran/blob/a7f67d4ad7ec5b45972c5aca3163e234abdc4db0/glotaran/utils/io.py#L166) under the hood. You could also use `load_datasets` and `Path.glob` in `load_data`.

Reply:
I do not think that this should be done.

The `import_dataset` mapping is only used once per project, thus this convinience has very little gain for users with the trade off of making the method sequence more complicated.

Another reason is that in future there might be more option to this like options to create an SVD or not and other preprocessing which should and must be set per dataset.

Last but not least: for users which just want to add a bulk of dataset can still simply copy them to the projects data folder. This is also the way to go if a user prefers to have e.g. the dataset as ascii wheres importing always converts to nc.

If it turns out that this functionality is really needed, we can add a method `import_multiple_data` in a different PR.

Reply:
Noted. Marking as resolved with a note to take these remarks into consideration in a future PR.